### PR TITLE
feat(flightsql): Support `CommandGetXdbcTypeInfo` metadata endpoint with an empty RecordBatch

### LINKED ISSUE #7569: FlightSQL Metadata: message `CommandGetXdbcTypeInfo`
The "correct" values for the returned data type info are driven by what the ODBC driver does, and we are taking the following steps to support `CommandGetXdbcTypeInfo`.

```[tasklist]
# Phases
- [x] 0. https://github.com/influxdata/influxdb_iox/pull/7662 that adds `CommandGetXdbcTypeInfo` in the Arrow DataFusion land
- [x] 1. https://github.com/influxdata/influxdb_iox/pull/7576
- [x] 2. https://github.com/influxdata/influxdb_iox/pull/7696 that supports `VARCHAR`, `INTEGER`, `FLOAT`, `TIMESTAMP`, and `INTERVALS`
- [ ] 3. Test with an ODBC client (Power BI)
- [ ] 4. Support the rest of [data types from DataFusion](https://arrow.apache.org/datafusion/user-guide/sql/data_types.html)
```

### PR DESCRIPTION
**Pull request by appletreeisyellow**
Closes #7569 

This PR is an initial support for FlightSQL `CommandGetXdbcTypeInfo` metadata endpoint which returns an empty RecordBatch.

Describe your proposed changes here.

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: flightsql/src/planner.rs [hunk 1]
```diff
@@ -417,6 +425,15 @@ async fn plan_get_table_types(ctx: &IOxSessionContext) -> Result<LogicalPlan> {
     Ok(ctx.batch_to_logical_plan(TABLE_TYPES_RECORD_BATCH.clone())?)
 }
 
+/// Return a `LogicalPlan` for GetXdbcTypeInfo
+async fn plan_get_xdbc_type_info(
+    ctx: &IOxSessionContext,
+    _data_type: Option<i32>,
+) -> Result<LogicalPlan> {
+    let batch = RecordBatch::new_empty(Arc::clone(&GET_XDBC_TYPE_INFO_SCHEMA));
```
Comment (alamb) [inline flightsql/src/planner.rs:433]: From my reading of the [spec](https://github.com/apache/arrow/blob/9588da967c756b2923e213ccc067378ba6c90a86/format/FlightSql.proto#L1064-L1113), the endpoint is supposed to return data about the actual sizes of the various types

Returning an empty record batch at first seems like a good start to me, but I suspect we'll have to start returning actual data eventually. 

If/when you get to the point you need to return information about the actual datatypes, I am happy to help figure out the values that should be returned

I think we may need to return a row for each value in https://github.com/apache/arrow/blob/9588da967c756b2923e213ccc067378ba6c90a86/format/FlightSql.proto#L948-L972

File: influxdb_iox/tests/jdbc_client/Main.java [hunk 2]
```diff
@@ -169,12 +169,15 @@ static void run_metadata(String url) throws SQLException {
...
 
-        // TODO uncomment when GetTables is implemented
-        //System.out.println("**************");
-        //System.out.println("getColumns:");
-        //System.out.println("**************");
-        //print_result_set(md.getColumns(null, null, null, null));
+        System.out.println("**************");
+        System.out.println("Type Info:");
+        System.out.println("**************");
+        print_result_set(md.getTypeInfo());
```
Comment (wolffcm) [inline influxdb_iox/tests/jdbc_client/Main.java:180]: I guess that these two calls `getColumns` and `getTypeInfo` are both using `CommandGetXdbcTypeInfo` under the hood?

File: influxdb_iox/tests/jdbc_client/Main.java [hunk 3]
```diff
@@ -169,12 +169,15 @@ static void run_metadata(String url) throws SQLException {
         System.out.println("**************");
         print_result_set(md.getTableTypes());
 
+        System.out.println("**************");
```
Comment (alamb) [inline influxdb_iox/tests/jdbc_client/Main.java:172]: 👍 

I think we should add the corresponding check of the output value here in the rust test (perhaps as as a follow on PR)

https://github.com/influxdata/influxdb_iox/blob/2341306c267c4a89a3b408d5b507dc733571f106/influxdb_iox/tests/end_to_end_cases/flightsql.rs#L1245-L1260

**Review by wolffcm — APPROVED** [review]
I guess you are just returning empty results for many of these methods to start.

I guess in the future it can return all the different kinds of data types that we support in DataFusion. I wonder if there is a DataFusion FlightSQL client that we could use for some of this?

**Review by alamb — APPROVED** [review]
Nice work @appletreeisyellow  -- I think this moves things along very nicely. 

I suspect we'll have to teach IOx to actually return values in this response, but just returning a response rather than unimplemented error is a step in the right direction 🚀 We can always add  actual datatype information to the response (perhaps if testing reveals it is needed)  as a follow on PR

**Comment by alamb** [comment]
whoops I see @wolffcm  had all the same points as me 😆  -- sorry about that

>  I wonder if there is a DataFusion FlightSQL client that we could use for some of this?

At the moment there is not one to my knowledge. It would be a great thing to add however, as I think several other DataFusion based projects are /have implemented FlightSQL

File: influxdb_iox/tests/jdbc_client/Main.java (hunk 2 above)
Comment (appletreeisyellow) [reply influxdb_iox/tests/jdbc_client/Main.java:180]: `getTypeInfo()` is using `CommandGetXdbcTypeInfo` under the hood, and `getColumns()` is a leftover piece from a previous implementation when `GetTables` is implemented. There was a TODO in the comment, so I just uncommented them in this PR

File: flightsql/src/planner.rs (hunk 1 above)
Comment (appletreeisyellow) [reply flightsql/src/planner.rs:433]: Totally, as the [ODBC support](https://github.com/influxdata/idpe/issues/17526) being tested, we might need the endpoint to return actual data soon

Comment (appletreeisyellow) [reply flightsql/src/planner.rs:433]: Follow up PR https://github.com/influxdata/influxdb_iox/pull/7696 with actual data type info returned

**Pull request merged**